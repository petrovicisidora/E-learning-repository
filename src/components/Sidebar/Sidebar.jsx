//Sidebar with image and course group name

import React from "react";

import {
  StyledSidebarContainer,
  StyledSidebarIcon,
  StyledSidebarTitle,
} from "./Sidebar.style.js";

function Sidebar({ image, title }) {
  return (
    <StyledSidebarContainer>
      <StyledSidebarIcon src={image} alt="Sidebar Icon" />
      <StyledSidebarTitle>{title}</StyledSidebarTitle>
    </StyledSidebarContainer>
  );
}

export default Sidebar;
