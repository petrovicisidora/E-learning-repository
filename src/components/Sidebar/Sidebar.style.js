import styled from "styled-components";

export const StyledSidebarContainer = styled.div`
  width: 100%;
  height: 70px;
  background-color: white;
  display: flex;
  align-items: center;
  margin-bottom: 10px;
  padding: 5px;
`;

export const StyledSidebarIcon = styled.img`
  margin-right: 20px;
  height: 60px;
  width: 60px;
`;

export const StyledSidebarTitle = styled.span`
  font-weight: bold;
`;
