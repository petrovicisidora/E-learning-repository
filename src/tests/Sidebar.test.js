import React from "react";
import { render, screen } from "@testing-library/react";

import Sidebar from "../components/Sidebar/Sidebar";

const sidebarProps = {
  image: "/path/to/image.png",
  title: "Test Sidebar",
};

test("renders sidebar title correctly", () => {
  render(<Sidebar {...sidebarProps} />);
  const sidebarTitle = screen.getByText(sidebarProps.title);

  expect(sidebarTitle).toBeInTheDocument();
});

test("renders sidebar icon correctly", () => {
  render(<Sidebar {...sidebarProps} />);
  const sidebarIcon = screen.getByAltText("Sidebar Icon");

  expect(sidebarIcon).toBeInTheDocument();
  expect(sidebarIcon).toHaveAttribute("src", sidebarProps.image);
});

test("renders sidebar title with correct styles", () => {
  render(<Sidebar {...sidebarProps} />);
  const sidebarElement = screen.getByText(sidebarProps.title);

  expect(sidebarElement).toHaveStyle("font-weight: bold");
});

test("renders sidebar icon with correct styles", () => {
  render(<Sidebar {...sidebarProps} />);
  const sidebarElement = screen.getByAltText("Sidebar Icon");

  expect(sidebarElement).toHaveStyle("height: 60px");
});

test("matches snapshot", () => {
  const { container } = render(<Sidebar />);
  expect(container).toMatchSnapshot();
});
