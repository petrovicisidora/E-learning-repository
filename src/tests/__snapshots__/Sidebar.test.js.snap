// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="sc-beySPh mfxcz"
  >
    <img
      alt="Sidebar Icon"
      class="sc-guDLey gmJwRK"
    />
    <span
      class="sc-dmyCSP fCRwGn"
    />
  </div>
</div>
`;
